osbenchmark/benchmark.py

```diff
@@ -863,6 +874,30 @@ def prepare_test_executions_dict(args, cfg):
                 test_executions_dict[execution] = None
     return test_executions_dict
 
+def run_and_aggregate(arg_parser, args, cfg):
+    semaphore = threading.Semaphore(1)
+    test_exes = []
+    args.subcommand = "execute-test"
+    aggregate_id = args.test_execution_id
+
+    for _ in range(args.test_iterations):
+        console.info(f"Running test {_ + 1}...")
+        args.test_execution_id = str(uuid.uuid4()) # we reuse the same args for each test so need to refresh the id
+        test_exes.append(args.test_execution_id)
+        with semaphore:
+            dispatch_sub_command(arg_parser, args, cfg)
```

Comment:
Is semaphore really needed? Wondering if there might be some conflicts with preexisting threading / Thespianpy within OSB. Might be overkilil. I think a simple for-loop should suffice?

Reply:
We should avoid using `dispatch_sub_command` and manipulating the args in line 880.

Reply:
Semaphore might be impacting the performance of each run. I ran a command that sets clients and target throughput to 10. This should ensure that final results have throughput of 10 ops/s. When I ran the command with these `auto-aggregate`, I see that target throughput is not being achieved across all runs and the aggregated results.

Command: `$ opensearch-benchmark auto-aggregate --test-iterations=3 --workload=geonames --target-hosts=XXXXXXX --client-options=basic_auth_user:'XXXXX',basic_auth_password:'XXXXXXX' --include-tasks="term" --test-mode --kill-running-processes --workload-params="search_clients:10,target_throughput:10"`
```
# Each test has the following results
|                                                  Segment count |        |         374 |        |
|                                                 Min Throughput |   term |       33.13 |  ops/s |
|                                                Mean Throughput |   term |       35.22 |  ops/s |
|                                              Median Throughput |   term |       35.69 |  ops/s |
|                                                 Max Throughput |   term |       36.39 |  ops/s |
```

However, when I run the command with `execute-test`, it reaches the target throughput.

Command: `$ opensearch-benchmark execute-test --workload=geonames --target-hosts=XXXXXXX --client-options=basic_auth_user:'XXXXXXX',basic_auth_password:'XXXXXXX--include-tasks="term" --kill-running-processes --workload-params="search_clients:10,target_throughput:10"`

```
|                                                  Segment count |        |         374 |        |
|                                                 Min Throughput |   term |        9.99 |  ops/s |
|                                                Mean Throughput |   term |          10 |  ops/s |
|                                              Median Throughput |   term |          10 |  ops/s |
|                                                 Max Throughput |   term |          10 |  ops/s |
```

Both commands were run on the same machine. We should reconsider using semaphore and go for a simpler approach. We can discuss this offline.

Reply:
> Is semaphore really needed? Wondering if there might be some conflicts with preexisting threading / Thespianpy within OSB. Might be overkilil. I think a simple for-loop should suffice?

I tried a simple for loop but i was having issues with already running processes 😕

Reply:
Discussed more offline and used an alternate approach. I tried a few test runs on the same machine with and without the changes, and the results were very similar. Let me know if you find any differences